src/components/label/label.tsx

```diff
@@ -80,6 +86,15 @@ export class Label {
   //
   //--------------------------------------------------------------------------
 
+  connectedCallback(): void {
+    this.calciteInternalLabelRegister.emit();
```

Comment:
Since you're listening on the document, why not dispatch the event from it too (similar to `disconnectedCallback`)?